### PR TITLE
[TS] LPP-45272 > LPS-154652 Convert .heic to .jpg for profile image

### PR DESCRIPTION
**Pull request by ces-hongvo**
Hi,
This PR is for the ticket https://issues.liferay.com/browse/LPP-45272. Please help me to review it and leave your comments.
Thank you.

PS: You will need to install the ImageMagick based on these instructions https://gist.github.com/hurricup/e14ae5bc47705fca6b1680e7a1fb6580 and https://issues.liferay.com/browse/LPS-151510 to test.
Update the portal-ext.properties with this value
mime.types.web.images=image/gif,image/jpeg,image/pjpeg,image/png,image/x-png,image/heic

### CODE REVIEW

File: modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/portlet/action/UploadImageMVCActionCommand.java [hunk 1]
```diff
@@ -204,6 +208,13 @@ private FileEntry _addTempImageFileEntry(PortletRequest portletRequest)
 		ThemeDisplay themeDisplay = (ThemeDisplay)portletRequest.getAttribute(
 			WebKeys.THEME_DISPLAY);
 
+		if (StringUtil.equalsIgnoreCase(mimeType, "image/heic")) {
+			file = _convertToJPEG(
+				file, UploadImageUtil.getMaxFileSize(portletRequest));
+			contentType = "image/jpeg";
```
Comment (tototrinh) [inline modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/portlet/action/UploadImageMVCActionCommand.java:214]: Please using ContentTypes.IMAGE_JPEG instead

File: modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/portlet/action/UploadImageMVCActionCommand.java [hunk 2]
```diff
@@ -204,6 +208,13 @@ private FileEntry _addTempImageFileEntry(PortletRequest portletRequest)
...
+			fileName = StringUtil.replaceLast(fileName, "heic", "jpg");
```
Comment (tototrinh) [inline modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/portlet/action/UploadImageMVCActionCommand.java:215]:
```
fileName = StringBundler.concat(
      FileUtil.stripExtension(fileName), StringPool.PERIOD,
      ImageTool.TYPE_JPEG);
```
would be better?

File: modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/portlet/action/UploadImageMVCActionCommand.java [hunk 3]
```diff
@@ -204,6 +208,13 @@ private FileEntry _addTempImageFileEntry(PortletRequest portletRequest)
 		ThemeDisplay themeDisplay = (ThemeDisplay)portletRequest.getAttribute(
 			WebKeys.THEME_DISPLAY);
 
+		if (StringUtil.equalsIgnoreCase(mimeType, "image/heic")) {
```
Comment (tototrinh) [inline modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/portlet/action/UploadImageMVCActionCommand.java:211]: "image/heic" should be added as a constants?

**Review by tototrinh — COMMENTED** [review]
@hongvo2308 I leave some comments please take a look and feel free to let me know if you have any questions. Thanks.

**Comment by ces-totrinh** [comment]
Forwarded: https://github.com/liferay-ts-reviewer/liferay-portal/pull/64
Thanks.

**Pull request closed without merging**